### PR TITLE
13-48 - created layout html/css for report cards and filter  

### PR DESCRIPTION
**Pull request by santosh-rwt**
Closes:-
1. [Filter Integration](https://vicesoftware.atlassian.net/browse/MAP-48)
2. [Card component with Graph](https://vicesoftware.atlassian.net/browse/MAP-13)

Update:-
1.  Added dashboard report cards.
2.  Added graph with cards.
3.  Added dashboard filter.

### CODE REVIEW

File: webapp/src/App.js [hunk 1]
```diff
@@ -1,19 +1,12 @@
...
 import './app.css'
-import NotificationPopup from './infrastructure/notificationPopup'
 
-function App() {
+const App = () => {
 	return (
 		<Router>
-			<NavBar />
-			<Container className='page'>
-				<NotificationPopup />
```
Comment (ghost) [inline webapp/src/App.js:14]: Don't remove the infrastructure code.

File: webapp/src/widgets/graph/Graph.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import { Line, defaults } from 'react-chartjs-2'
+
+defaults.global.defaultFontSize = '8'
```
Comment (ghost) [inline webapp/src/widgets/graph/Graph.js]: Why is this necessary? And what it does?

Comment (santosh-rwt) [reply webapp/src/widgets/graph/Graph.js]: it is used to set labels fontsize

File: webapp/src/widgets/cards/images/icon/arrow-down-dark.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<svg width="149" height="87" viewBox="0 0 149 87" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (sahilvicesoftware) [inline webapp/src/widgets/cards/images/icon/arrow-down-dark.svg]: follow camelCase in file names

Comment (sahilvicesoftware) [inline webapp/src/widgets/cards/images/icon/arrow-down-dark.svg]: i think in PR #1 we had a diff folder for icons ?

File: webapp/src/features/dashboard/DashboardFilter.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+						selected={endDate}
+						onChange={(endDate) => setEndDate(endDate)}
+						dateFormat='MMMM d, yyyy'
+						className='form-control calender-input'
+					/>
+				</Form.Group>
+				<Form.Group className='mb-15'>
+					<Form.Label>plan</Form.Label>
+					<Form.Control as='select'>
+						<option>all plans</option>
```
Comment (sahilvicesoftware) [inline webapp/src/features/dashboard/DashboardFilter.js]: should be All Plans or ALL PLANS

File: webapp/src/features/dashboard/DashboardFilter.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+					</Form.Control>
+				</Form.Group>
+				<Form.Group className='mb-15'>
+					<Form.Label>subscription status</Form.Label>
+					<Form.Control as='select'>
+						<option>all status</option>
```
Comment (sahilvicesoftware) [inline webapp/src/features/dashboard/DashboardFilter.js]: All Status or ALL STATUS

File: webapp/src/features/dashboard/DashboardFilter.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+					<Form.Label>end date</Form.Label>
+					<DatePicker
+						selected={endDate}
+						onChange={(endDate) => setEndDate(endDate)}
+						dateFormat='MMMM d, yyyy'
+						className='form-control calender-input'
+					/>
+				</Form.Group>
+				<Form.Group className='mb-15'>
+					<Form.Label>plan</Form.Label>
```
Comment (sahilvicesoftware) [inline webapp/src/features/dashboard/DashboardFilter.js]: if form.label is visible - it should be Plan

File: webapp/src/features/dashboard/DashboardFilter.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+					<Form.Control as='select'>
+						<option>all plans</option>
+					</Form.Control>
+				</Form.Group>
+				<Form.Group className='mb-15'>
+					<Form.Label>subscription status</Form.Label>
```
Comment (sahilvicesoftware) [inline webapp/src/features/dashboard/DashboardFilter.js]: if form.label is visible - it should be Subscription Status

File: webapp/src/features/dashboard/DashboardFilter.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import React, { useState } from 'react'
+import { Col, Button, Form } from 'react-bootstrap'
+import Cards from '../../widgets/cards'
+import DatePicker from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import './DashboardFilter.css'
+
```
Comment (sahilvicesoftware) [inline webapp/src/features/dashboard/DashboardFilter.js]: rename the component to DashboardFilters

File: webapp/src/widgets/graph/Graph.js [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+	return (
+		<div className='graph'>
+			<Line
+				data={data}
+				options={{
+					responsive: true,
+					maintainAspectRatio: false,
+					scales: {
+						xAxes: [
+							{
```
Comment (sahilvicesoftware) [inline webapp/src/widgets/graph/Graph.js]: Axes should be Axis

**Review by sahilvicesoftware — CHANGES_REQUESTED** [review]

File: webapp/src/widgets/cards/images/icon/arrow-down-dark.svg (hunk 3 above)
Comment (ghost) [reply webapp/src/widgets/cards/images/icon/arrow-down-dark.svg]: updated

Comment (ghost) [reply webapp/src/widgets/cards/images/icon/arrow-down-dark.svg]: updated

File: webapp/src/features/dashboard/DashboardFilter.js (hunk 6 above)
Comment (ghost) [reply webapp/src/features/dashboard/DashboardFilter.js]: updated

File: webapp/src/features/dashboard/DashboardFilter.js (hunk 7 above)
Comment (ghost) [reply webapp/src/features/dashboard/DashboardFilter.js]: updated

File: webapp/src/widgets/graph/Graph.js (hunk 9 above)
Comment (ghost) [reply webapp/src/widgets/graph/Graph.js]: This is property defined in react-chartjs 2, so will not be appropriate to change the name.

File: webapp/src/features/dashboard/DashboardFilter.js (hunk 4 above)
Comment (ghost) [reply webapp/src/features/dashboard/DashboardFilter.js]: Actually need to display 'ALL PLANS' which is handled using css uppercase property.

File: webapp/src/features/dashboard/DashboardFilter.js (hunk 5 above)
Comment (ghost) [reply webapp/src/features/dashboard/DashboardFilter.js]: Actually need to display 'ALL STATUS' which is handled using css uppercase property.
